content/docs/glossary/sponsored-reserves.mdx

```diff
@@ -85,8 +58,248 @@ See [Revoke Sponsorship](../start/list-of-operations.mdx#revoke-sponsorship) for
 
 The logic above does not detail any of the error cases, which are specified [here](../start/list-of-operations.mdx#revoke-sponsorship).
 
-#### Examples
+## Examples
+Each example builds on itself, referencing variables from previous snippets. We'll demonstrate a few different things you can do with sponsoring:
+ - Sponsor creation of a trustline for another account.
+ - Sponsor **two** trustlines for an account via two *different* sponsors.
+ - Transfer sponsorship responsibility from one account to another.
+ - Revoke sponsorship by an account entirely.
+
+(For brevity, we'll assume the existence of a `SignAndSend(...)` method (defined [below](#footnote)) which will creates and submits a transaction with the proper parameters and error-checking.
+
+### Preamble
+We'll start by including the boilerplate of account and asset creation.
+
+<CodeExample>
+
+```go
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	sdk "github.com/stellar/go/clients/horizonclient"
+	"github.com/stellar/go/keypair"
+	"github.com/stellar/go/network"
+	protocol "github.com/stellar/go/protocols/horizon"
+	"github.com/stellar/go/txnbuild"
+)
+
+func main() {
+	client := sdk.DefaultTestNetClient
+
+	// Both S1 and S2 will be sponsors for A at various points in time.
+	S1, A, S2 := keypair.MustRandom(), keypair.MustRandom(), keypair.MustRandom()
+	sponsoredAccount := A.Address()
+
+	for _, pair := range []*keypair.Full{S1, A, S2} {
+		resp, err := http.Get("https://friendbot.stellar.org/?addr=" + pair.Address())
+		if err != nil {
+			panic(err)
+		}
+		resp.Body.Close()
+		fmt.Println("Funded", pair.Address())
+	}
+
+	//
+	// NOTE: Error checks are omitted for brevity; always check return values!
+	//
+
+	// Load the corresponding account for both A and C.
+	s1Account, err := client.AccountDetail(sdk.AccountRequest{AccountID: S1.Address()})
+	aAccount, err := client.AccountDetail(sdk.AccountRequest{AccountID: A.Address()})
+	s2Account, err := client.AccountDetail(sdk.AccountRequest{AccountID: S2.Address()})
+
+	// Arbitrary assets to sponsor trustlines for. Let's assume they make sense.
+	assets := []txnbuild.CreditAsset{
+		txnbuild.CreditAsset{Code: "ABCD", Issuer: S1.Address()},
+		txnbuild.CreditAsset{Code: "EFGH", Issuer: S1.Address()},
+		txnbuild.CreditAsset{Code: "IJKL", Issuer: S2.Address()},
+	}
+}
+```
+
+</CodeExample>
+
+
+### Sponsoring a Trustline
+Now, let's sponsor trustlines for Account A. Notice how the `CHANGE_TRUST` operation is sandwiched between the begin and end sponsoring operations and that all relevant accounts need to sign the transaction.
+
+<CodeExample>
+
+```go
+	//
+	// 1. S1 will sponsor a trustline for Account A.
+	//
+	sponsorTrustline := []txnbuild.Operation{
+		&txnbuild.BeginSponsoringFutureReserves{
+			SourceAccount: &s1Account,
+			SponsoredID:   sponsoredAccount,
```

Comment:
Should `sponsoredAccount` be `aAccount` here? If so, the other uses of `sponsoredAccount` should probably be updated to the corresponding account as well.

Reply:
`aAccount` is the full key-pair, whereas `sponsoredAccount` is just its public key (you can see `sponsoredAccount := aAccount.Address()` happen earlier). I figured it was a more descriptive variable name, but would you say it's more confusing since it implies a difference between the two?

Reply:
Ah I see. Yeah I was a little confused about what was going on here, but that's my fault for not clearly reading the preamble. I think the example would be easier to understand if you just used `A.Address()` though.  That line already specifies that you're setting the `SponsoredID`, which is just another way to say sponsored account. You could add a comment to clarify that. What do you think?

Reply:
I think you're right. I was trying to figure out why I went with a separate variable in the first place and it's because of [this line](https://github.com/stellar/new-docs/pull/254/files/9fc83ec4eb9901ff885f87929bc5b33967a33a90#diff-1973c43d48bf44e72fbc4e404a4dbb22R206) (and other revocations). The `Account:` field should be a `*string` type, but you can't pass `&A.Address()` (I'm not sure why, pretty new to Go myself; I think it's because it's an rvalue?). 

I think renaming the variable to `A := aAccount.Address()` will kill two birds with one stone: consolidate all references to the public key *and* make it clear that it's still Account A, not some newly-introduced sponsored account. Thoughts?

Reply:
Yeah that sounds like a good idea!

Reply:
Had to use `aAddress` since `A` was taken lol but hopefully still gets the point across.